### PR TITLE
chore(identify): remove deprecated symbols

### PR DESCRIPTION
**Pull request by thomaseizinger**
## Change checklist

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] A changelog entry has been made in the appropriate crates

### CODE REVIEW

File: protocols/identify/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 - Add `Info` to the `libp2p-identify::Event::Pushed` to report pushed info.
   See [PR 4527](https://github.com/libp2p/rust-libp2p/pull/4527)
+- Remove deprecated `initial_delay`.
+  Identify requests are always sent instantly after the connection has been established.
+  See [PR XXXX](https://github.com/libp2p/rust-libp2p/pull/XXXX)
```
Comment (thomaseizinger) [inline protocols/identify/CHANGELOG.md]:
```suggestion
  See [PR 4735](https://github.com/libp2p/rust-libp2p/pull/4735)
```

**Comment by thomaseizinger** [comment]
Needs https://github.com/libp2p/rust-libp2p/pull/4735 to fix kademlia tests.

**Comment by mxinden** [comment]
> Needs #4735 to fix kademlia tests.

:scream: we are doomed. A recursive dependency.

**Comment by thomaseizinger** [comment]
> > Needs #4735 to fix kademlia tests.
> 
> 😱 we are doomed. A recursive dependency.

Hahaha. I meant #4743.

**Comment by thomaseizinger** [comment]
@mxinden I almost introduced a bug here where we waited for the interval initially before sending the first identify. I added a test to cover that.

**Comment by mxinden** [comment]
> @mxinden I almost introduced a bug here where we waited for the interval initially before sending the first identify. I added a test to cover that.

Much appreciated!

**Pull request merged**